### ISSUE TITLE
can't use `from hypothesis import *`: Item in ``from list'' not a string

### ISSUE THREAD

**Issue by radix** [DRMacIver/hypothesis#112]
I can't do `from hypothesis import *`. This is because `__init__.py` uses unicode_literals. It seems the only string literals in `__init__.py` are the strings in the `__all__` list, which need to be `str` / `bytes` on python2, so I think it should be safe to just disable that import in that file.

```
radix@伝説:~$ mktmpenv
New python executable in tmp-49ba2df47c475d58/bin/python2.7
Also creating executable in tmp-49ba2df47c475d58/bin/python
Installing setuptools, pip, wheel...done.
This is a temporary environment. It will be deleted when you run 'deactivate'.

(tmp-49ba2df47c475d58)radix@伝説:~/.virtualenvs/tmp-49ba2df47c475d58$ pip install hypothesis
You are using pip version 7.0.3, however version 7.1.0 is available.
You should consider upgrading via the 'pip install --upgrade pip' command.
Collecting hypothesis
Installing collected packages: hypothesis
Successfully installed hypothesis-1.9.0
(tmp-49ba2df47c475d58)radix@伝説:~/.virtualenvs/tmp-49ba2df47c475d58$ python -c 'from hypothesis import *'
Traceback (most recent call last):
  File "<string>", line 1, in <module>
TypeError: Item in ``from list'' not a string
```

**Comment by radix** [comment]
Oh, or you could use `b'foo'` syntax.

**Comment by radix** [comment]
Nope, never mind. They _can't_ be bytes on python3, so I think the only solution is to disable `unicode_literals` in the `__init__.py` files.

**Comment by radix** [comment]
... or to decode everything in it explicitly on python 2? or use six.text_type? I guess hypothesis doesn't depend on six. I notice that `searchstrategies.py` also defines a buggy `__all__`, so disabling unicode_literals will be difficult there.

... this is probably one of the reasons why unicode_literals is usually avoided :)

**Comment by radix** [comment]
I was going to to work on a PR but I will let a maintainer of the project decide the strategy to go with :)

**Comment by DRMacIver** [comment]
Ugh. Thanks for the report. Basically all the options with things that expect a str in both python 2 and 3 are terrible. The best solution is probably going to be to explicitly cast everything to str in \_\_all\_\_, which is why I have to do in a few other places, but that will make the linter unhappy.

Full disclosure: I don't approve of Python 2 or * imports, so given this is a pain to fix I probably wouldn't recommend holding your breath. I'm not saying I won't fix it - I almost certainly will, but there's a whole bunch of stuff prioritised in front of it so if you want this to work you're probably better off submitting a pull request. :-)

**Comment by radix** [comment]
I'm not particularly concerned about `import *` either (though Py2 is another story). I only noticed it because I sometimes use it at the REPL when I'm tinkering around to save myself from too much typing.

**Comment by DRMacIver** [comment]
This is now fixed in master. I'm waiting on some unrelated things before I ship a new release, but there should be one out fairly soon.
